File: app/src/main/java/com/example/kotlin/java/ApplyTest.java
package com.example.kotlin.java;

import android.content.Context;
import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

/**
 * @author zhangquan
 */
public class ApplyTest {
    public void test(Context ctx) {
        TextView textView = new TextView(ctx);
        textView.setText("text");
        textView.setTextColor(Color.parseColor("#ffffff"));
        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
    }
}

File: app/src/main/java/com/example/kotlin/kotlin/inline/ApplyTest.kt
package com.example.kotlin.kotlin.inline

import android.content.Context
import android.graphics.Color
import android.view.View
import android.widget.TextView

/**
 * 初始化对象并赋值，返回对象本身
 * @author zhangquan
 */
class ApplyTest {
    fun test(ctx: Context?) {
        //初始化对象并赋值，返回对象本身
        val textView = TextView(ctx).apply{
            text="text"
            setTextColor(Color.parseColor("#ffffff"))
            setOnClickListener{
            }
        }
        //正常写法
        textView.setOnClickListener(object :View.OnClickListener{
            override fun onClick(v: View?) {

            }
        })
        //只有一个方法的lambda写法
        textView.setOnClickListener {

        }

    }
}